scipy/signal/_fir_filter_design.py

```diff
@@ -426,26 +432,27 @@ def firwin(numtaps, cutoff, *, width=None, window='hamming', pass_zero=True,
                              f'{pass_zero}')
     pass_zero = bool(operator.index(pass_zero))  # ensure bool-like
```

Comment:
Side note: According to the docstr, `pass_zero in (True, False, 'bandpass', 'lowpass', 'highpass', 'bandstop')` needs to hold.
Couldn't the line 
```python
pass_zero = bool(operator.index(pass_zero))
``` 
be replaced with
```python
if isinstance(pass_zero, str):
   ...
elif pass_zero not in (True, False)):
    raise ValueError(...)
```
to make the code a little more accessible?

Reply:
Great idea! Not sure what `...` would be though. Would you add a patch? (push to my branch or drop a patch here for me to apply or fix in a follow-up, whichever is easier).

Reply:
Thank you! PR merged into this branch in https://github.com/scipy/scipy/pull/22343/commits/6c30b7d0ef8200c39c497b01a40eafe21d358139